### ISSUE TITLE
model.val(data="yolo_seg/data.yaml", split='val', save_json=True)生成的 .json文件乱码

### ISSUE THREAD

**Issue by ggyybb** [ultralytics/ultralytics#18709]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

![Image](https://github.com/user-attachments/assets/ca746748-255c-4930-9c96-4659ddd6b13d)
like this.

### Environment

Ultralytics 8.3.54 🚀 Python-3.8.17 torch-1.10.1+cu102 CUDA:0 (NVIDIA GeForce MX330, 2048MiB)

### Minimal Reproducible Example

from ultralytics import YOLO

if __name__ == '__main__':
    model = YOLO('train_result/yolo_trained_weights/weights/best.pt')
    results = model.val(data="yolo_seg/data.yaml", split='val', save_json=True)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ggyybb, thank you for reporting this issue and for your interest in Ultralytics 🚀! We recommend checking out our [Docs](https://docs.ultralytics.com) for helpful tips and resources. If you are encountering issues, these are often addressed in our [Python](https://docs.ultralytics.com/usage/python/) or [CLI](https://docs.ultralytics.com/usage/cli/) usage guides.

Regarding your 🐛 bug report, thank you for including a minimal reproducible example (MRE) 🙌! This helps us a lot in debugging the issue. An Ultralytics engineer will review your report soon and assist further. In the meantime, here are some steps you can try to troubleshoot the problem: 

## Upgrade 

Ensure you are running the latest version of `ultralytics` and dependencies to verify the issue isn't already resolved. Upgrade using:

```bash
pip install -U ultralytics
```

You might also want to confirm your environment is properly configured with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), such as [**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

YOLO works in several verified environments, which include:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Steps for Reporting Issues

If possible, please ensure that the `.json` file produced is using a valid text encoding (e.g., UTF-8). You may also want to verify whether the issue is related to specific data or configurations within your `val` split.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

Ensure you check our CI status. If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing, verifying the correct operation of YOLO across platforms.

For further assistance, feel free to join the Ultralytics community:
- Chat with us on [Discord](https://discord.com/invite/ultralytics) 🎧
- Discuss on [Discourse](https://community.ultralytics.com)
- Share ideas on [Reddit](https://reddit.com/r/Ultralytics)

This is an automated response, and an Ultralytics engineer will offer additional support soon. Thank you for your patience! 😊

**Comment by ambitious-octopus** [comment]
Hey @ggyybb, this isn't a bug. The segmentation masks are encoded, so to make them human-readable, you'll need to decode them. You can use the following script to achieve this:

```python
import numpy as np
import json
from pycocotools.mask import decode
import cv2
model = YOLO("yolo11n-seg.pt")
results = model.val(data="coco8-seg.yaml", save_json=True)
json_path = results.save_dir / "predictions.json"

with open(json_path, "r") as f:
    data = json.load(f)
    

instance = 0
mask = data[instance]["segmentation"]

def single_decode(rle):
    rle['counts'] = rle['counts'].encode('utf-8')
    mask = decode(rle)
    return np.asarray(mask, dtype=np.uint8)

decoded = single_decode(mask) * 255

cv2.imshow("mask", decoded)
cv2.waitKey(0)
cv2.destroyAllWindows()
```

**Comment by ambitious-octopus** [comment]
LLM-generated Explanation of the Script:

Model Initialization and Validation:
The script begins by loading a YOLO model with segmentation capabilities using the YOLO class and a pre-trained weights file (yolo11n-seg.pt).
It then validates the model using a dataset configuration file (coco8-seg.yaml) and stores the results in a JSON file.

File Handling:
The path to the generated predictions JSON file is defined, and the file is opened to load the segmentation data.

Data Processing:

For each segmentation instance in the JSON data, the script retrieves the encoded mask.

Decoding Function:
A helper function single_decode is defined to decode the Run-Length Encoded (RLE) mask using the pycocotools library. The RLE counts are encoded to bytes to ensure compatibility with the decode function.
The decoded mask is converted into a NumPy array of type uint8.

Visualization:
The decoded mask is scaled by multiplying it by 255 to make it more visible when displayed as an image.
OpenCV is used to display each decoded mask, and it waits for a key press to close the display window.

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/18136

**Comment by ambitious-octopus** [comment]
@Y-T-G, i think we need to add this to the documentation. Do you have any idea where to put it?

**Comment by Y-T-G** [comment]
@ambitious-octopus We could add it. We don't have task-specific argument description though.

https://docs.ultralytics.com/modes/val/

**Comment by Y-T-G** [comment]
@ambitious-octopus We can. We don't have task-specific description for arguments. Just the one under Val docs.

https://docs.ultralytics.com/modes/val/#arguments-for-yolo-model-validation

`save_json` has different outputs for each of the tasks, so it would be a good idea to document it. Probably needs a new section under the docs for each of the tasks
